apps/emqx_connector/src/mqtt/emqx_connector_mqtt_worker.erl

```diff
@@ -245,238 +325,112 @@ pre_process_conf(Key, Conf) ->
-            "_as_'egress'_is_not_configured",
-        messages => Msg
-    });
-do_send_async(
+import_msg(
     #{
-        connection := Connection,
-        mountpoint := Mountpoint,
-        connect_opts := #{forwards := Forwards}
+        dup := Dup,
+        payload := Payload,
+        properties := Props,
+        qos := QoS,
+        retain := Retain,
+        topic := Topic
     },
-    Msg,
-    Callback
+    #{server := Server}
 ) ->
-    Vars = emqx_connector_mqtt_msg:make_pub_vars(Mountpoint, Forwards),
-    ExportMsg = emqx_connector_mqtt_msg:to_remote_msg(Msg, Vars),
-    ?SLOG(debug, #{
-        msg => "publish_to_remote_broker",
-        message => Msg,
-        vars => Vars
-    }),
-    emqx_connector_mqtt_mod:send_async(Connection, ExportMsg, Callback).
-
-disconnect(#{connection := Conn} = State) when Conn =/= undefined ->
-    emqx_connector_mqtt_mod:stop(Conn),
-    State#{connection => undefined};
-disconnect(State) ->
-    State.
-
-format_mountpoint(undefined) ->
-    undefined;
-format_mountpoint(Prefix) ->
-    binary:replace(iolist_to_binary(Prefix), <<"${node}">>, atom_to_binary(node(), utf8)).
-
-name(Id) -> list_to_atom(str(Id)).
+    #{
+        id => emqx_guid:to_hexstr(emqx_guid:gen()),
+        server => Server,
+        payload => Payload,
+        topic => Topic,
+        qos => QoS,
+        dup => Dup,
+        retain => Retain,
+        pub_props => printable_maps(Props),
+        message_received_at => erlang:system_time(millisecond)
+    }.
 
-obfuscate(Map) ->
+printable_maps(undefined) ->
+    #{};
+printable_maps(Headers) ->
     maps:fold(
-        fun(K, V, Acc) ->
-            case is_sensitive(K) of
-                true -> [{K, '***'} | Acc];
-                false -> [{K, V} | Acc]
-            end
+        fun
+            ('User-Property', V0, AccIn) when is_list(V0) ->
+                AccIn#{
+                    'User-Property' => maps:from_list(V0),
+                    'User-Property-Pairs' => [
+                        #{
+                            key => Key,
+                            value => Value
+                        }
+                     || {Key, Value} <- V0
+                    ]
+                };
+            (K, V0, AccIn) ->
+                AccIn#{K => V0}
         end,
-        [],
-        Map
+        #{},
+        Headers
     ).
 
-is_sensitive(password) -> true;
-is_sensitive(ssl_opts) -> true;
-is_sensitive(_) -> false.
-
-str(A) when is_atom(A) ->
-    atom_to_list(A);
-str(B) when is_binary(B) ->
-    binary_to_list(B);
-str(S) when is_list(S) ->
-    S.
+%% TODO
+% maybe_destroy_session(#{connect_opts := ConnectOpts = #{clean_start := false}} = State) ->
+%     try
+%         %% Destroy session if clean_start is not set.
+%         %% Ignore any crashes, just refresh the clean_start = true.
+%         _ = do_connect(State#{connect_opts => ConnectOpts#{clean_start => true}}),
+%         _ = disconnect(State),
+%         ok
```

Comment:
This also needs some attention, @zmstone. As discussed before I got an impression that this _should not_ be needed anymore? Does it also mean that we should drop the [`clean_start`](https://github.com/emqx/emqx/blob/78028a7fcf26a64fe10290348c853cac26b3521e/apps/emqx_connector/src/mqtt/emqx_connector_mqtt_schema.erl#L108-L115) field from the bridge config so that a user wouldn't be able to set it to `false`?

Reply:
#9889